### DISCUSSION TITLE
Accessing Muted Files for Research: Dumping In-Memory Mutants (if Possible)?

### DISCUSSION THREAD

**Discussion by Ali94k** [stryker-mutator/stryker-net#2937] [Q&A]
I'm conducting research that requires access to muted files and analyze them. As far as I know mutants are kept in-memory. 

Here's my question:
Is there a way to dump these in-memory mutants into a file format suitable for analysis?

I'm open to exploring alternative approaches as well. If modifying the source code to facilitate mutant export is an option, I'd be happy to contribute with some guidance.

I appreciate any insights or suggestions you might have.

**Comment by rouke-broersma** [comment]
Mutants can be exported as json with the json reporter. Doesn't that fit your needs?

**Reply by Ali94k** [reply]
Thanks, dev-mode and Json exporter are what I needed

**Comment by dupdob** [comment]
What kind of format are you expecting ?
source files?

**Reply by Ali94k** [reply]
ideally something like the file below.

https://github.com/Flash0ver/F0-Talks-MutationTesting/blob/main/resources/FizzBuzzBuilder.Stryker.cs

**Reply by dupdob** [reply]
Thanks
In the curren Stryker version, you can use `--dev-mode' CLI option: the mutated source code will be part of the logs: 
```
Mutated MyFile.cs:
.... mutated code
```
It still requires to extract it from the log files.

Adding an option to dump/export the mutated code can be discussed. As it is trickier than it looks for the general case, the first thing to understand is what is the intended use of it, and what are the general requirements: should it output net projects (that compile) or only source code? if it is only source code, what is the expected (folder) structure ? Which code variant should be dumped : the raw mutated one or the one Stryker uses for testing after removing 'compilation error' mutants ?

Note that the actual source code will be a bit more messy than your example as Stryker generates rather long and random namespace to avoid any naming conflicts.

**Reply by Ali94k** [reply]
Thanks! For now, `dev-mode` and the JSON exporter meet my requirements. I'm still in the early stages of my research, but I may share any new requirements that arise as I progress.
